client/my-sites/plans/current-plan/jetpack-checklist/index.tsx

```diff
@@ -241,12 +287,19 @@ export default connect(
 			  } )
 			: undefined;
 
+		const planSlug = getSitePlanSlug( state, siteId );
+		const isPremium = !! planSlug && isPremiumPlan( planSlug );
+		const isProfessional = ! isPremiumPlan && !! planSlug && isBusinessPlan( planSlug );
```

Comment:
Did you mean to check for `! isPremium` instead of `! isPremiumPlan`?

Reply:
That will help, thanks 😅

Reply:
Applied in d3b39c40bbe23176fab173fe8b650dffb0454723, fixes the missing tasks for different plans you mentioned.